In this project i wil analize the sales of houses in a city in Australia. For this i will use the raw\_sales dataset that iincludes the following variables:

Datesold: The date when an owner sold the house to a buyer.

Postcode: 4 digit postcode of the suburb where the owner sold the property

Price: Price for which the owner sold the property.

With this data i will solve th following tasks

- Which date corresponds to the highest number of sales?
    
- Find out the postcode with the highest average price per sale? 
    
- Which year witnessed the lowest number of sales?
    
- Use a query  to deduce the top six postcodes by year's price.

First i will take a look at the data

In [2]:
SELECT top 10 *
from raw_sales

(10 rows affected)

Total execution time: 00:00:00.133

datesold postcode price propertyType bedrooms 2007-02-07 00:00:00.0000000 2607 525000 house 4 2007-02-27 00:00:00.0000000 2906 290000 house 3 2007-03-07 00:00:00.0000000 2905 328000 house 3 2007-03-09 00:00:00.0000000 2905 380000 house 4 2007-03-21 00:00:00.0000000 2906 310000 house 3 2007-04-04 00:00:00.0000000 2905 465000 house 4 2007-04-24 00:00:00.0000000 2607 399000 house 3 2007-04-30 00:00:00.0000000 2606 1530000 house 4 2007-05-24 00:00:00.0000000 2902 359000 house 3 2007-05-25 00:00:00.0000000 2906 320000 house 3

In [4]:
select count(*) as Amount_of_null_values
from raw_sales
where postcode is null or price is null or propertyType is null or bedrooms is null

(1 row affected)

Total execution time: 00:00:00.120

(No column name) 0

Now as we can see the dataset doesnt contain any null values, so it is possible to continue using the table without any modifications

To solve the first question we need to analyze the data that hast the most amount of sales. This will be solved for a year, then for a month and in a specific day

In [4]:
Select year(datesold) as Year, count(*) as Cantidad
from raw_sales
group by year(datesold)
order by count(*) DESC

(13 rows affected)

Total execution time: 00:00:00.143

Año Cantidad 2017 4541 2016 3908 2018 3858 2015 3648 2014 2863 2013 2119 2012 1858 2011 1633 2010 1555 2009 1426 2019 1385 2008 639 2007 147

In [5]:
SELECT MONTH(datesold) as Month, count(*) as Cantidad
from raw_sales
group by MONTH(datesold)
order by count(*) DESC

(12 rows affected)

Total execution time: 00:00:00.144

MES Cantidad 11 3102 3 2878 5 2797 2 2656 10 2548 4 2468 12 2432 8 2429 7 2386 9 2367 6 2213 1 1304

In [10]:
select top 10 datesold, count(*) as Amount_of_sales
from raw_sales
group by datesold
order by count(*) DESC

(10 rows affected)

Total execution time: 00:00:00.157

datesold (No column name) 2017-10-28 00:00:00.0000000 50 2017-11-18 00:00:00.0000000 39 2018-03-24 00:00:00.0000000 38 2017-04-08 00:00:00.0000000 37 2017-11-11 00:00:00.0000000 37 2018-02-24 00:00:00.0000000 35 2017-02-25 00:00:00.0000000 34 2017-11-04 00:00:00.0000000 34 2019-07-01 00:00:00.0000000 34 2015-11-07 00:00:00.0000000 33

From the three querys we can say some tthings about the amount of sales. First when talking about year, 2017 was the year that has the most amount of sales, and that almost every year the amount of sales have been increasing with the expection of 2019

Now, regarding the month, November it is by a margin the month with the most amount of sales, then it follows March and May

Finally the 28 of october from 2017 was by far the day that has the most amount of sales, with a difference of 11 to the second date. The year is expected considering the amount of sales that have been done.

> Now it comes a question about what is the post code with the highest price. In order to find this we need to do the following query

In [9]:
SELECT  top 10 postcode, price
from raw_sales
order by price DESC

(10 rows affected)

Total execution time: 00:00:00.137

postcode price 2611 8000000 2603 7300000 2611 6480000 2600 5475000 2912 5425000 2603 4700000 2603 4500000 2600 4417000 2603 4220000 2902 4150000

in the matter of finding the year wit the least amount of sales we need to do the next query

This means that 8.000.000  is the highest price and that property is linked to this postal code 2611

In [12]:
select year(datesold) as Year, count(*) as Amount_of_sales
from raw_sales
group by year(datesold)
order by count(*)

(13 rows affected)

Total execution time: 00:00:00.139

Year Amount_of_sales 2007 147 2008 639 2019 1385 2009 1426 2010 1555 2011 1633 2012 1858 2013 2119 2014 2863 2015 3648 2018 3858 2016 3908 2017 4541

This means that 2017 was the year with the least amount of sales with 147, four times less than 2008

Next, in order to find the top six post code by year we need to do the following query

In [55]:
with top_postcode as (SELECT postcode, price, Year(datesold) as Year,
RANK() OVER (PARTITION BY Year(datesold) ORDER BY Price DESC) AS Rank
FROM raw_sales)
SELECT postcode, price, Year
from top_postcode
where Rank <=6
order by Year, price DESC

(78 rows affected)

Total execution time: 00:00:00.162

postcode price Year 2603 1780000 2007 2606 1530000 2007 2603 1460000 2007 2612 1105000 2007 2913 1100000 2007 2615 930000 2007 2603 3500000 2008 2603 1425000 2008 2600 1380000 2008 2603 1375000 2008 2602 1300000 2008 2603 1250000 2008 2603 2600000 2009 2603 2200000 2009 2603 2100000 2009 2612 1875000 2009 2604 1620000 2009 2603 1580000 2009 2603 7300000 2010 2604 2425000 2010 2603 2400000 2010 2600 2350000 2010 2603 1780000 2010 2603 1700000 2010 2912 5425000 2011 2902 4150000 2011 2606 2500000 2011 2603 2375000 2011 2606 2200000 2011 2600 2100000 2011 2603 2760000 2012 2600 1980000 2012 2606 1900000 2012 2603 1730000 2012 2603 1650000 2012 2600 1605000 2012 2600 1880000 2013 2603 1650000 2013 2600 1620000 2013 2600 1510000 2013 2600 1501000 2013 2603 1495000 2013 2603 3400000 2014 2603 3400000 2014 2603 3160000 2014 2603 2600000 2014 2603 2550000 2014 2603 2080000 2014 2611 8000000 2015 2603 2660000 2015 2600 2450000 2015 2603 2440000 2015 2603 2400000 2015 2603 2322000 2015 2611 6480000 2016 2600 5475000 2016 2600 4417000 2016 2600 3950000 2016 2603 3625000 2016 2603 2595000 2016 2603 4700000 2017 2603 4220000 2017 2603 3500000 2017 2603 3160000 2017 2603 3050000 2017 2603 3000000 2017 2600 3750000 2018 2603 2950000 2018 2603 2900000 2018 2604 2800000 2018 2603 2740000 2018 2600 2701000 2018 2603 4500000 2019 2603 2195000 2019 2603 1975000 2019 2606 1900000 2019 2602 1770000 2019 2605 1701000 2019

This last query shows the top 6 postal code by price and year and price. So it shows the most expensive houses. As we can see some postcodes appears over and over wich is expected considering the fact that this are the most expensive houses, so when they have been sold they appear in the highest places.